test/auth.js

```diff
@@ -18,16 +19,62 @@ describe('auth interface', () => {
         .then((resp) => should(resp.uri).startWith('safe-auth:'));
   });
 
+  it('throws error if missing permissions object', async () => {
+    const app = h.createTestApp();
+    const test = () => app.auth.genAuthUri();
+    should(test).throw(errConst.MISSING_CONTAINERS_OBJECT.msg);
+  });
+
+  it('throws error if permissions object contains invalid permission', async () => {
+    const app = h.createTestApp();
+    const test = () => app.auth.genAuthUri({ _public: ['Invalid'] });
+    should(test).throw("'Invalid' is not a valid permission");
+  });
+
+  it('should throw error if non-standard container is requested', () => {
+    const containersPermissions = { _app: ['Read', 'Insert', 'ManagePermissions'] };
+    const result = createAuthenticatedTestApp('_test_scope', containersPermissions, { own_container: true });
+    should(result).be.rejectedWith(`
+    thread '<unnamed>' panicked at '
+
+    !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+    !   unwrap! called on Result::Err                                              !
+    !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+    src\\test_utils.rs:89,23 in safe_app::test_utils
+
+    Err(
+        Unexpected(
+                "'_app' not found in the access container"
+                    )
+                    )
+
+                    ', C:\\Users\\viv\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\unwrap-1.1.0\\src\\lib.rs:67:24
```

Comment:
this all seems _very_ specific.

to windows/whitespace/viv

Reply:
yeh that guy always has issues lol. Not even sure where this came from. Don't remember me submitting anything with hardcoded paths :|

+1 to generic asserts ofc.